src/metatrain/utils/metrics.py

```diff
@@ -23,32 +24,42 @@ def update(self, predictions: Dict[str, TensorMap], targets: Dict[str, TensorMap
         """
 
         for key, target in targets.items():
-            if key not in self.information:
-                self.information[key] = (0.0, 0)
             prediction = predictions[key]
+            for block_key in target.keys:
+                target_block = target.block(block_key)
+                prediction_block = prediction.block(block_key)
 
-            for prediction_block, target_block in zip(
-                prediction.blocks(), target.blocks()
-            ):
+                key_to_write = key
+                if self.separate_blocks:
+                    for name, value in zip(block_key.names, block_key.values):
+                        key_to_write += f"_{name}_{int(value)}"
```

Comment:
so this would create something like `density_o3_lambda_2_o3_sigma_-1`? Would it make sense to try to format it as `density (o3_lambda=2, o3_sigma=-1)` instead? 

And as a similar improvement for gradients, should we change `energy_positions_gradients` to `energy gradients w.r.t. positions`? This way the two of them together would be `density (o3_lambda=2, o3_sigma=-1) gradients w.r.t. positions`?

Reply:
For the first case, `LabelsEntry.__str__` should get you pretty close already.

Reply:
IMO we should remove the whitespaces so it's easier to read (visually) and also parse
e.g. `density(o3_lambda=2;o3_sigma=-1)`

The gradients are always re-formatted anyway for the user. `energy_positions_gradients` will never be shown

Reply:
> IMO we should remove the whitespaces so it's easier to read (visually) and also parse e.g. density(o3_lambda=2;o3_sigma=-1)

I don't think parsing these is a good idea. If we really need this information, we should just save the loss as a TensorMap somewhere.

I don't mind too much about removing the whitespace if you prefer, but I don't see the point: these are for human consumption, so they should be made easy to read by humans. I would keep `,` instead of `;` though, since it is already used in a couple of other places.

> The gradients are always re-formatted anyway for the user.

But what about something like density gradients?

Reply:
The classic (and pretty much only) way to plot training curves is by parsing log files

Reply:
> The classic (and pretty much only) way to plot training curves is by parsing log files

That's not great TBH. I would rather have an extra output file containg the loss at all epochs instead of forcing parsing for no reason

Reply:
Ok, opening issue